source/framework/vx_producer.c

```diff
@@ -0,0 +1,1336 @@
+        }
+    }
+    if (NULL == found_ref)
+    {
+        VX_PRINT(VX_ZONE_INFO, "PRODUCER no buffer found with status %d \n", status);
+    }
+
+    return found_ref;
+}
+
+static vx_int32 get_buffer_id(vx_reference current_ref, vx_producer producer)
+{
+    vx_int32 buffer_id;
+    for (buffer_id = 0; (vx_uint32)buffer_id < producer->numBufferRefs; buffer_id++)
+    {
+        if (producer->refs[buffer_id].ovx_ref == current_ref)
+        {
+            VX_PRINT(VX_ZONE_REFERENCE, "PRODUCER found a buffer ref %p at index %d \n", current_ref, buffer_id);
+            break;
+        }
+    }
+
+    if ((vx_uint32)buffer_id == producer->numBufferRefs)
+    {
+        buffer_id = -1;
+        VX_PRINT(
+            VX_ZONE_ERROR,
+            "PRODUCER Dequeued reference cannot be found in the consumer registered references %s",
+            "\n");
+    }
+
+    return buffer_id;
+}
+
+#ifdef IPPC_SHEM_ENABLED
+static void check_ippc_clients_connected(vx_producer producer)
+{
+    // if one of the receiver is ready, register it and the sender can start sending data
+    for (vx_uint32 i = 0U; i < OVXGW_NUM_CLIENTS; i++)
+    {
+        if (E_IPPC_OK == ippc_sender_receiver_ready(&producer->m_sender_ctx.m_sender, i) && 
+            (producer->consumers_list[i].state == PROD_STATE_CLI_NOT_CONNECTED))
+        {
+            producer->consumers_list[i].state       = PROD_STATE_CLI_CONNECTED;
+            producer->consumers_list[i].consumer_id = i;
+            producer->nb_consumers++;
+        }
+    }    
+}
+
+static void fill_reference_info(vx_producer producer, SProducerContent* buffid_message)
+{
+    buffid_message->num_refs = producer->numBufferRefsExport;
+    uint32_t i = 0;
+    uint32_t j = 0;
+    for (i = 0; i < producer->numBufferRefsExport; i++)
+    {
+        vx_enum                  ref_type;
+        vx_uint32                num_items          = 0;
+        tivx_utils_ref_ipc_msg_t ipc_message_parent = {0};
+        tivx_utils_ref_ipc_msg_t ipc_message_item[OVXGW_MAX_NUM_REFS];
+
+        vx_status framework_status =
+            vxQueryReference(producer->refs[i].ovx_ref, VX_REFERENCE_TYPE, (void*)&ref_type, (vx_size)sizeof(ref_type));
+        if (framework_status != VX_SUCCESS)
+        {
+            VX_PRINT(VX_ZONE_ERROR, "PRODUCER: vxQueryReference() failed for object [%d]\n", i);
+            break;
+        }
+        else if (ref_type == VX_TYPE_OBJECT_ARRAY)
+        {
+            framework_status = rbvx_utils_export_ref_for_ipc_xfer_objarray(
+                producer->refs[i].ovx_ref,
+                &num_items,
+                &ipc_message_parent,
+                (tivx_utils_ref_ipc_msg_t*)&ipc_message_item[0]);
+
+            buffid_message->num_items = num_items;
+
+            // send object array items data, if present
+            for (j = 0; j < num_items; j++)
+            {
+                memcpy(&buffid_message->ipc_msg[i][j], (void*)&ipc_message_item[j], sizeof(tivx_utils_ref_ipc_msg_t));
```

Comment:
don't memcopy if you do need it: you have the pointer to the memory buffid_message where you have to write into. So pass this point to the export function.

Reply:
Done